feat: added initial rework of docs

0xcadams opened the pull request:
Updated docs with:
1. Revamped search using [cmdk](https://github.com/pacocoursey/cmdk/tree/main) and lazy imports instead of a fetch to pull the JSON. Reduced the debounce time and added preloading so it feels fast. Fixed markdown to plaintext.
2. Reworked navigation: renamed sections for (hopefully) more clarity, added icons & animations, changed grouping, and made some groups default to closed to feel less overwhelming.
3. Random rewording for clarity.
4. ~~Added fun hotkeys for quick navigation around the docs.~~

<img width="2528" alt="Screenshot 2025-07-09 at 4 41 07 PM" src="https://github.com/user-attachments/assets/a2f50e55-34b2-4f03-8fde-ad65d5c0ec05" />
<img width="2506" alt="Screenshot 2025-07-09 at 4 42 03 PM" src="https://github.com/user-attachments/assets/7841dd09-a296-431a-adb4-a35fb9398dac" />
<img width="2443" alt="Screenshot 2025-07-09 at 4 41 48 PM" src="https://github.com/user-attachments/assets/45f7aeb6-b60b-413b-a89b-ea4bb414f3f3" />
<img width="2523" alt="Screenshot 2025-07-09 at 4 41 38 PM" src="https://github.com/user-attachments/assets/e09b3eb8-05fb-493c-b750-7b74c7f6872e" />

aboodman commented:
Wow!

Supportive of most of this. Specific feedback:

* I do not like the visual noise of the hotkeys on the buttons and links. Sorry.
* I do not like to use rules (lines) as separators in our designs if we can avoid it. I like the layouts to be clean enough to not need them. I do not think they are needed here. Remove.
* I like the improved performance of the search.
* I do not like how the search box changes size depending on how many results there are. It should always be the same size.
* I am mixed on the icons, but I want to try it out for a bit.
* I like the restructure of the nav, but I think that "Postgres" should continue being the second section
* There is a layout bug introduced around the subnav
 on the right side where it doesn't always align w/ the search box (see video: https://github.com/user-attachments/assets/82f980df-d102-4199-9db7-8cfa16b837fe)
* "markdown to text"  –  nice fix.
* I need to go through the copy changes.

alexhking commented:
Nice! This is a whole lot of work and looks pretty cool. I did notice a few things looking through the code and while testing it out. Going to do a deep dive and then do inline comments, etc. Thanks!

0xcadams commented:
Sounds great - I will make updates based on this (and Alex I'll wait for your feedback). If either of you have any changes for how the icons are shown, let me know. I was trying to make the sections clearer visually.

I will make smaller PRs in the future, got a bit ahead of myself.

alexhking commented:
https://github.com/user-attachments/assets/a9288d45-dd0f-4f2b-ae62-6bc5a0ef0b65

@0xcadams ☝🏼 Initial visual review

alexhking commented on app/page.tsx:
```diff
@@ -57,21 +72,26 @@ export default function Home() {
           <Button
-            variant="outline"
-            className={styles.primaryButton}
+            className="flex items-center gap-2"
```
I notice that you're putting back the tailwind classes. Initially when we created this docs site from Aria Docs, we wanted to get rid of tailwind completely. I'm not a huge fan of it, but if you're more comfortable using it we can keep it.

alexhking commented on components/search.tsx:
```diff
@@ -289,65 +251,89 @@ export default function Search() {
+                    <CommandItem
+                      key={item.id}
+                      value={item.id}
+                      onSelect={() => {
```
This is where things could break with standard browser behavior. Handling navigation with router.push breaks standard link functionality.

alexhking commented on app/globals.css:
```diff
@@ -110,18 +111,14 @@
+  padding: 0.2rem;
+  margin: 0 0.05rem;
+  background-color: hsl(var(--primary-highlight));
+  font-weight: 500;
```
Why have you added a duplicate font-weight declaration here? (see line 116)

alexhking commented on components/Pagination.tsx:
```diff
@@ -1,46 +1,75 @@
           >
-            <span className="flex items-center text-muted-foreground text-xs">
-              <ChevronLeftIcon className="w-[1rem] h-[1rem] mr-1" />
+            <span className="flex items-center text-muted-foreground gap-2 text-sm">
```
I noticed you removed the left and right arrows here. They could be unnecessary, but I did think they provided quick visual clarity for users.

0xcadams replied:
I added these back and moved the hotkeys to tooltip. Agreed it looks better that way

alexhking commented on components/search.tsx:
```diff
@@ -194,82 +197,41 @@ export default function Search() {
-  }, [isOpen, highlightIndex, searchResults]);
+  // Toggle the menu when ⌘K is pressed
+  useHotkeys(
+    'meta+k',
```
What to ensure that this still works for non-mac users (Ctrl+K on Windows)

0xcadams replied:
Thanks for the callout - fixed this to use `mod`: https://react-hotkeys-hook.vercel.app/docs/documentation/useHotkeys/basic-usage#modifiers--special-keys

alexhking commented on components/toc.tsx:
```diff
@@ -1,10 +1,46 @@
+    [tocs, level2Tocs],
+  );
+
+  useHotkeys(['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'], e => {
```
What do we do if more than 10 headings exist?

0xcadams replied:
The hotkeys only apply to the first ten - and if there's only one header it doesn't show them.

alexhking commented on components/toc.tsx:
```diff
@@ -1,10 +1,46 @@
 import {ActiveHashLink} from './ui/ActiveHashLink';
+import Kbd from './ui/kbd';
+
+export default async function Toc({
```
Could be wrong here, but I thought React didn't support async function components on the client? (You added 'use client' above)

0xcadams replied:
Yeah, good call. This was actually the source of the bug you found. I thought next didn't allow this at all

alexhking reviewed (commented):
Added some inline comments along with the video from above. Thanks for the nice work!

0xcadams commented:
Thanks for the feedback!

I made some changes to the search box - pinned to the top. It's more similar to how it currently is - let me know if this doesn't address feedback.

> * I do not like the visual noise of the hotkeys on the buttons and links. Sorry.

I moved this to a tooltip instead of showing in the button. Agreed it was noisy.

> * I do not like to use rules (lines) as separators in our designs if we can avoid it. I like the layouts to be clean enough to not need them. I do not think they are needed here. Remove.

Removed

> * I am mixed on the icons, but I want to try it out for a bit.

Open to any changes here

> * I like the restructure of the nav, but I think that "Postgres" should continue being the second section

Done

> * There is a layout bug introduced around the subnav
>   on the right side where it doesn't always align w/ the search box (see video: https://github.com/user-attachments/assets/82f980df-d102-4199-9db7-8cfa16b837fe)

Fixed this

> * I need to go through the copy changes.

Should've done these separately, will do this in the future.

Alex:
* I changed the icon for the solidjs - let me know if this is better. The icons are mostly based on Lucide (shadcn and a lot of projects use it): https://lucide.dev/icons. Exceptions are the brand logos (React and SolidJS)
* I fixed the back button issue - thanks for finding that.

aboodman commented:
Few more bits of visual feedback in attached movie:
 CleanShot 2025-07-12 at 00.23.31.mp4
<https://drive.google.com/file/d/18jRcLQwQ-NR6yX_fQ8qIsTvpDGXSYvJK/view?usp=drive_web>
As for tooltips, sorry still not a fan, it just seems superfluous. How
often is somebody going to click on these? Are shortcuts really needed? It
seems harder to process that there is a shortcut and use it than to just
click?

Is this something you yourself want Chase? Or thinking others wants it?




On Fri, Jul 11, 2025 at 10:43 AM Alexander King ***@***.***>
wrote:

> *alexhking* left a comment (rocicorp/zero-docs#148)
> <https://github.com/rocicorp/zero-docs/pull/148#issuecomment-3063762823>
>
> Thanks for the quick revisions. On the surface it looks good to me, the
> hover states are cool. Curious what @aboodman
> <https://github.com/aboodman> thinks. Otherwise, lgtm.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rocicorp/zero-docs/pull/148#issuecomment-3063762823>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAATUBGBO2CRID3WDQA4BED3IAOWJAVCNFSM6AAAAACBE3HVZGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTANRTG43DEOBSGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

aboodman commented:
> I assume a lot of early adopters using Zero (and early inspirational products using sync engines) have keyboard shortcuts. To my mind, it's more of a reflection of how zero users want their products to be - fast to load, fast to search, and have keyboard shortcuts for fast navigation.

Ah! That's an interesting rationale. But yeah I think let's not do things artificially. The docs are a reflection of our quality standards too and part of quality is making things useful (and only making things that are useful).

Anyway, we have zbugs which is an actual real app that we can use and add keyboard shortcuts to.

aboodman commented:
A trick I often use with product questions is just asking what we ourselves want. We are the target market for Zero.

0xcadams commented:
Sounds great and that's a good rule of thumb. It is a bit artificial. I will remove!

0xcadams commented:
@aboodman unfortunately the recording you shared doesn't seem to have any sound. I think I can infer that spacing needs to be adjusted but it's hard to infer the rest.

0xcadams commented:
Just updated this with:
 - fixed space between the header and the welcome and introduction
 - remove height animation for the search
 - organization of docs: postgres after concepts
 
 Let me know if you want to see anything else @aboodman

aboodman commented:
🎉

On Wed, Jul 16, 2025 at 4:55 PM Chase Adams ***@***.***>
wrote:

> Merged #148 <https://github.com/rocicorp/zero-docs/pull/148> into main.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rocicorp/zero-docs/pull/148#event-18670965504>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAATUBDUJ7SPWL5TRNL6YQL3I4GAVAVCNFSM6AAAAACBE3HVZGVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJYGY3TAOJWGU2TANA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
